src/Sarif/Query/Evaluators/ResultEvaluators.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Sarif.Query.Evaluators
+{
+    public static class SarifEvaluators
+    {
+        public static IExpressionEvaluator<Result> ResultEvaluator(TermExpression term)
+        {
+            switch (term.PropertyName.ToLowerInvariant())
+            {
+                case "baselinestate":
+                    return new EnumEvaluator<Result, BaselineState>(r => r.BaselineState, term);
+                case "correlationguid":
+                    return new StringEvaluator<Result>(r => r.CorrelationGuid, term, StringComparison.OrdinalIgnoreCase);
+                case "guid":
+                    return new StringEvaluator<Result>(r => r.Guid, term, StringComparison.OrdinalIgnoreCase);
+                case "hostedvieweruri":
+                    return new StringEvaluator<Result>(r => r.HostedViewerUri?.ToString(), term, StringComparison.OrdinalIgnoreCase);
+                case "kind":
+                    return new EnumEvaluator<Result, ResultKind>(r => r.Kind, term);
+                case "level":
+                    return new EnumEvaluator<Result, FailureLevel>(r => r.Level, term);
```

Comment:
It's a bit weird that the severity level is an EnumEvaluator. But I guess there's no choice since Level is defined as an enum in SARIF?

Reply:
Yep. The evaluator for each property is dictated by the type in the code. Enum => EnumEvaluator.